semgrep_output_v1.atd

```diff
@@ -1113,29 +1114,35 @@ type scan_request = {
 }
 
 (* Response by the backend to the CLI to the POST /scans *)
-(* TODO
 type scan_response = {
-    scan: scan_response_bis;
+    scan: scan_info;
+    scan_config: scan_configuration;
+    engine_config: engine_configuration;
 }
-type scan_response_bis = {
-    (* the scan id *)
-    id: string;
-    (* since 1.37.0 *)
-    ?enabled_products: product list;
+
+(* meta info about the scan *)
+type scan_info = {
+    id: int; (* the scan id *)
+    deployment_id: int;
+    deployment_name: string;
+}
+
+(* config specific to the scan, eg *)
+type scan_configuration = {
+    enabled_products: product list;
+    rules: raw_json; (* can we type this better *)
+    ~ignored_files: string list;
+    ~triage_ignored_syntactic_ids: string list;
+    ~triage_ignored_match_based_ids: string list;
 }
-*)
 
-(* TODO? seems used only for dry_run and passed urlencoded to /scans/config?xxx
-type scan_config_request = {
-    dry_run: bool;
-    (* project_metadata.repository *)
-    repo_name: string;
-    (* TODO: deprecate, prefer scan_request.scan_metadata.requested_products *)
-    sca: bool;
-    full_scan: bool;
-    semgrep_version : version;
+(* settings for the cli *)
+type engine_configuration = {
```

Comment:
I'm not sure about the split between scan_configuration and engine_configuration.
Maybe better to merge them. Those are both engine configuration; What is an engine config vs scan config?

Reply:
In fact I started with zach this new 'ci_config_from_cloud' that tries to be the more general and new way
to send scan config configs from the cloud to the CLI. Wou;ld be great to deprecate
this engine_configuration and rely on ci_config_from_cloud instead.

Reply:
We should probably chat about this more. I know you just added `ci_config_from_cloud` but I really want to get rid of it and replace it with this.

The intention behind the design here is that `scan_config` encapsulates all the state that changes scan to scan, and is tightly coupled with policy/rule selection for a particular repo. This gets built by calling out to various products `generate_config` functions and assembled. In the future, we may want to handle this differently in the backend.

`engine_config` on the other hand is relatively static, gets generated per deployment, with some possibility for per repo overrides in the future, but is ultimately product agnostic. 

fyi @zyannes

Reply:
Also, I should mention that "ci" is just one kind of scan and I'm trying to work towards unifying the APIs used by `semgrep scan` and `semgrep ci` so that's another reason why I don't want CI in the name of the type.

Reply:
actually we want to differentiate more scan and ci with @milanwilliams and @zzeleznick  so I'm not sure 'semgrep scan' should  do anything with the backend. It should be mostly for local use.

Reply:
But yes we should chat more about ci_config_from_cloud and who does what.